restorer.py

```diff
@@ -56,15 +56,15 @@ def restore_db_backup(self):
         Target:
             - restore a database's backup in PostgreSQL.
         '''
-        replicator = Replicator(self.connecter, self.new_dbname,
-                                Default.RESTORING_TEMPLATE, self.logger)
-        result = self.connecter.allow_db_conn(Default.RESTORING_TEMPLATE)
-        if result:
-            replicator.replicate_pg_db()
-            self.connecter.disallow_db_conn(Default.RESTORING_TEMPLATE)
-        else:
-            self.logger.stop_exe(Messenger.ALLOW_DB_CONN_FAIL.format(
-                dbname=Default.RESTORING_TEMPLATE))
+        #replicator = Replicator(self.connecter, self.new_dbname,
+                                #Default.RESTORING_TEMPLATE, self.logger)
+        #result = self.connecter.allow_db_conn(Default.RESTORING_TEMPLATE)
+        #if result:
+            #replicator.replicate_pg_db()
+            #self.connecter.disallow_db_conn(Default.RESTORING_TEMPLATE)
+        #else:
+            #self.logger.stop_exe(Messenger.ALLOW_DB_CONN_FAIL.format(
+                #dbname=Default.RESTORING_TEMPLATE))
```

Comment:
Those are the lines which create a new database before restoring the selected backup (they call replicator because actually the new database is a replica of template0).

Those lines are working, they create a database with the desired name, but for any reason, despite being before the restoring process, the new database isn't being recognized and the restorer stops the execution with the error "The database X does not exist".

After the error, executing the restorer again will result in another error, but this time in the replicator (cannot create the database with the desired name because it already exists).

So, for the moment, I comment that code to avoid this problem, and make the user create the database manually before restoring it.